flow/layers/physical_shape_layer.h

```diff
@@ -37,11 +37,18 @@ class PhysicalShapeLayer : public ContainerLayer {
 
 #if defined(OS_FUCHSIA)
   void UpdateScene(SceneUpdateContext& context) override;
+
+  void UpdateSceneChildren(SceneUpdateContext& context);
 #endif  // defined(OS_FUCHSIA)
 
   float total_elevation() const { return total_elevation_; }
 
  private:
+#if defined(OS_FUCHSIA)
+  float local_scenic_elevation_ = 0.0f;
```

Comment:
#2 and #3 in this list should be unnecesary

Reply:
see comments above for child_layer_exists_below_

As far as children_need_system_compositing_ -- I think you are right, we can just get away with always calling UpdateSceneChildren in PhysicalShapeLayer::UpdateScene. I initially thought I needed it for correctness, but I think at this point it's just a tiny optimization. Getting rid of it will simplify things.

Reply:
got rid of #1 and #3 as per our discussion.